blivet/devicetree.py

```diff
@@ -150,12 +150,19 @@ def _add_device(self, newdev, new=True):
             :param newdev: the device to add
             :type newdev: a subclass of :class:`~.devices.StorageDevice`
 
-            Raise ValueError if the device's identifier is already
+            Raise DeviceTreeError if the device's identifier is already
             in the list.
         """
         if newdev.uuid and newdev.uuid in [d.uuid for d in self._devices] and \
            not isinstance(newdev, NoDevice):
-            raise ValueError("device is already in tree")
+            # Just found a device with already existing UUID. Is it the same device?
+            dev = self.get_device_by_uuid(newdev.uuid, incomplete=True, hidden=True)
+            if dev.name == newdev.name:
+                raise DeviceTreeError("Trying to add already existing device.")
+            else:
+                raise DeviceTreeError("Duplicate disk UUID '%s' found for devices: '%s' and '%s'.\n\
+                                       This is usually caused by cloning the device image resulting \
```

Comment:
I would leave out the work 'disk' here, since there could be other types of UUID that get duplicated when cloning a device. It seems to usually be a disklabel UUID, but it could be something else as well.

Reply:
Fixed.